decoder.rb

```diff
@@ -0,0 +1,53 @@
+morse_dict = {
+  '.-' => 'A',
+  '-...' => 'B',
+  '-.-.' => 'C',
+  '-..' => 'D',
+  '.' => 'E',
+  '..-.' => 'F',
+  '--.' => 'G',
+  '....' => 'H',
+  '..' => 'I',
+  '.---' => 'J',
+  '-.-' => 'K',
+  '.-..' => 'L',
+  '--' => 'M',
+  '-.' => 'N',
+  '---' => 'O',
+  '.--.' => 'P',
+  '--.-' => 'Q',
+  '.-.' => 'R',
+  '...' => 'S',
+  '-' => 'T',
+  '..-' => 'U',
+  '...-' => 'V',
+  '.--' => 'W',
+  '-..-' => 'X',
+  '-.--' => 'Y',
+  '--..' => 'Z',
+  '-----' => '0',
+  '.----' => '1',
+  '..---' => '2',
+  '...--' => '3',
+  '....-' => '4',
+  '.....' => '5',
+  '-....' => '6',
+  '--...' => '7',
+  '---..' => '8',
+  '----.' => '9',
+  '.-.-.-' => '.',
+  '--..--' => ',',
+  '..--..' => '?'
+}
+
+define_method :decode do |code|
+  words = code.split('   ')
+  words.each do |word|
+    characters = word.split
+    characters.each do |character|
+      print morse_dict[character]
+    end
+    print ' '
+  end
+end
+decode('-- -.--   -. .- -- .')
```

Comment:
- [ ] Based on the project requirements, you are expected to have at least three methods with a single responsibility  - e.g 
    - `decode_char` - responsibility is to decode a single character. 
    - `decode_word` - responsibility is to decode a word and 
    - `decode` - responsibility is to decode a sentence. 
Then, use the **character decode method** inside the **word decode method** and so on.

_**Please note that the names of the methods can be different from what I call them above but each name should be meaningful at first glance.**_ 


Project requirement 👇🏾 
---- |
![image](https://user-images.githubusercontent.com/84156714/232233641-7e1252d6-eda0-44cf-af06-b01cc307b1db.png)


Please carefully read the requirement again and implement this project accordingly.

Reply:
Thanks for sharing this information, I will create the `decode_char`, `decode_word` and `decode` methods